dashboard/test/models/level_test.rb

```diff
@@ -464,6 +464,34 @@ def create_maze
     assert_equal 'PlantASeed', level.name
   end
 
+  test 'cannot create two blockly levels with same key' do
+    game = create :game, name: 'Artist'
+    level1 = create :artist, name: 'blockly', level_num: '5_5_5', user_id: nil, game: game
+    assert_equal 'blockly:Artist:5_5_5', level1.key
+    e = assert_raises ActiveRecord::RecordInvalid do
+      create :artist, name: 'blockly', level_num: '5_5_5', user_id: nil, game: game
+    end
+    assert_includes e.message, 'Validation failed: Level num has already been taken'
+  end
+
+  test 'cannot create multi and match levels with same key' do
+    multi = create :multi, name: 'multi match', level_num: nil, user_id: nil
+    assert_equal 'multi match', multi.key
+    e = assert_raises ActiveRecord::RecordInvalid do
+      create :match, name: 'multi match', level_num: nil, user_id: nil
+    end
+    assert_includes e.message, 'Validation failed: Name has already been taken'
+  end
```

Comment:
I was able to verify that the above test would fail previously and passes due to the changes in this PR.